src/main/java/com/google/devtools/build/lib/rules/java/JavaOptions.java

```diff
@@ -597,11 +610,25 @@ public ImportDepsCheckingLevelConverter() {
               + "--java_header_compilation is enabled.")
   public boolean requireJavaToolchainHeaderCompilerDirect;
 
+  private Label getHostJavaBase() {
+    if (hostJavaBase == null) {
+      if (useRemoteJdkAsHostJavaBase
+          // If there's no embedded JDK, we did use the local JDK as host JDK in the past.
+          // In order to make this flag flip as soft as possible, we will continue to do so.
```

Comment:
To confirm: there are no known reasons to prefer using a local JDK in this case, and the motivation for this logic is to make this change a direct replacement for the current behaviour?

Any concerns with following up with another change (maybe behind another `--incompatible_` flag) to remove this part? Being able to rely on the supported `--host_javabase` version for Bazel binaries without an embedded JDK seems like a Feature.

Reply:
Yes, my motivation is to have a direct replacement for current behavior.

I guess we should figure out what the long term plan is before we the other flag, i.e. whether we want to default to the local JDK for everything or the remote JDK.

Reply:
Okay. I'm not sure the additional complexity here is buying us much, but I understand wanting to reduce the risk of this change. How is this logic being tested? Is there coverage in `bazel_java_test.sh` for the no-JDK case?

Reply:
The `"1".equals(System.getProperty("embedded_jdk")` part is going away after we flip the flag?

Is the removal of `"1".equals(System.getProperty("embedded_jdk")` going to happen behind a different `--incompatible_changes_` flag?

Reply:
The whole check that we do in Java code now will go away once we flip/remove the flag since we'll do that in the BUILD file instead.

The property is bazel internal and not exposed to any of the bazel users, is it? So we can just remove it.

Reply:
It results in different behaviour for the binaries with and without embedded JDKs, which is visible to users. Maybe I'm just not understanding the plan.

> The whole check that we do in Java code now will go away once we flip/remove the flag since we'll do that in the BUILD file instead.

What will the replacement for `"1".equals(System.getProperty("embedded_jdk")` be when that happens?

Reply:
> It results in different behaviour for the binaries with and without embedded JDKs, which is visible to users. Maybe I'm just not understanding the plan.


Yes, the behavior is different for users with and without embedded JDK. It is already different today and I don't want to make it the same in this PR. This is basically maintaining current behavior.


> What will the replacement for "1".equals(System.getProperty("embedded_jdk") be when that happens?

external/embedded_jdk links to the local JDK if there's no embedded JDK if I'm not mistaken. We'll use that then.

Reply:
> > > The whole check that we do in Java code now will go away once we flip/remove the flag since we'll do that in the BUILD file instead.
> >
> > What will the replacement for "1".equals(System.getProperty("embedded_jdk") be when that happens?
>
> external/embedded_jdk links to the local JDK if there's no embedded JDK if I'm not mistaken. We'll use that then.

I mean, how will that logic be expressed? You mentioned something about putting it in the BUILD file instead of current Java code?

Maybe it would be helpful to see the follow-up patch you have in mind, if I'm not following the explanation.

Reply:
Thanks for pushing on it. 
Turns out that we have to keep some logic about the embedded JDK if we don't want to switch the no JDK case. Example commit:
https://github.com/meisterT/bazel/commit/5a2c55afee8d5e2a8d75c5f7b5c327820e3944d1

So we have two options:
- keep this check and don't change how we handle the host JDK in the no JDK case 
- remove the check and also switch how we handle the host JDK in the no JDK case

Preferences?

Reply:
As discussed offline with @lberki, the plan is to try the "remove the check and also switch how we handle the host JDK in the no JDK case" option. Everything else here LGTM.